### PR TITLE
fix: do not reload entities after saving #182829

### PR DESCRIPTION
**Pull request by OwenLacey28**
| Action | Done? |
| --- | --- |
| CHANGELOG updated | ✔ |
| Code compiles and unit tests all pass locally | ✔ |
| Debug/console log code removed | ✔ |
| Commented out code removed | ✔ |
| Unit tests added/updated | ✔ |
| README updated | N/A - bugfixes |
| Considered: <br /> - Performance <br /> - Security <br /> - Logging | ✔ |
| Licenses of any new/upgraded dependencies checked, with no copyleft dependencies introduced | ✔ |

### CODE REVIEW

File: src/Audacia.Seed.EntityFrameworkCore/Repositories/EntityFrameworkCoreSeedableRepository.cs [hunk 1]
```diff
@@ -533,8 +533,6 @@ private T Seed<T>(EntitySeed<T> seed) where T : class
         }
 
         Add(entity);
-        // Reload the entity so that the state is `Unchanged` rather than `Detached` and we can subsequently use this entity for setting up other relationships.
-        _afterSaveJobs.Add(() => _context.Entry(entity).Reload());
```
Comment (OwenLacey28) [inline src/Audacia.Seed.EntityFrameworkCore/Repositories/EntityFrameworkCoreSeedableRepository.cs:537]: this is the 'change'. The rest of it is making the tests work

File: src/Audacia.Seed/Customisation/EntitySeedExtensions.cs [hunk 2]
```diff
@@ -40,7 +39,17 @@ public static EntitySeed<TEntity> With<TEntity, TProperty>(
                 $"The provided {nameof(getter)} ({getter}) does not access a property on {typeof(TEntity).Name}.");
         }
 
-        entitySeed.AddCustomisation(new SeedPropertyConfiguration<TEntity, TProperty>(getter, value));
+        // Check the getter passed in is setting a nested property or not
+        // E.g x => x.Parent.GrandParentId is nested, but x => x.ParentId is not.
+        if (getter.Body is MemberExpression m && m.Expression?.Type != typeof(TEntity))
```
Comment (OwenLacey28) [inline src/Audacia.Seed/Customisation/EntitySeedExtensions.cs:44]: made these work like the existing WithDifferent method for nesting properties

File: docs/adr/0001-clear-the-ef-core-change-tracker.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# 1. Clear the EF Core change tracker
```
Comment (Audacia-RhysSmith) [inline docs/adr/0001-clear-the-ef-core-change-tracker.md:1]: Thanks for adding this, helps explain the changes

**Review by Audacia-RhysSmith — APPROVED** [review]

**Review by Audacia-RhysSmith — CHANGES_REQUESTED** [review]
image
Just ran the tests and got this, is this something from the change?

**Comment by OwenLacey28** [comment]
> image Just ran the tests and got this, is this something from the change?

this is what happens when you run against a real SQL db - so it's expected for you to see this at the start of your first test run

**Review by Audacia-RhysSmith — APPROVED** [review]

**Pull request merged**